cedar-policy-core/src/parser/err.rs

```diff
@@ -48,6 +48,76 @@ pub enum ParseError {
     #[error("poorly formed: {0}")]
     #[diagnostic(code(cedar_policy_core::parser::to_ast_err))]
     ToAST(String),
+
+    /// Error in the CST -> AST transform, found an illegal, empty node.
+    #[error("poorly formed: {0}")]
+    #[diagnostic(code(cedar_policy_core::parser::to_ast_err))]
+    NodeShouldNotBeEmpty(String),
```

Comment:
I think this may be a type of `InternalInvariantViolation` since I imagine that the message "node should not be empty" reflects an error on our end.